src/hbbft_bba.erl

```diff
@@ -366,10 +370,16 @@ rand_val(2#11) -> hd(hbbft_utils:random_n(1, [0, 1])).
 val(2#1) -> 0;
 val(2#10) -> 1.
 
-add_witness(Id, Value, Witness) ->
+add_witness(Id, Value, Witness, AllowUnion) ->
     Old = maps:get(Id, Witness, 0),
     New = add(Value, Old),
     case Old /= New of
+        true when New == 3, AllowUnion == false ->
```

Comment:
I don't _quite_ understand the difference here if both the true clauses return the same Witness. Perhaps I'm missing something...

Reply:
The guards are different, I suppose we could combine them but they represent distinct cases.